llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -10077,19 +10077,46 @@ static void checkMixedPrecision(Loop *L, OptimizationRemarkEmitter *ORE) {
   }
 }
 
-static bool areRuntimeChecksProfitable(GeneratedRTChecks &Checks,
-                                       VectorizationFactor &VF, Loop *L,
-                                       PredicatedScalarEvolution &PSE,
-                                       ScalarEpilogueLowering SEL,
-                                       std::optional<unsigned> VScale) {
+static InstructionCost calculateEarlyExitCost(LoopVectorizationCostModel &CM,
+                                              VPlan &Plan, ElementCount VF) {
+  InstructionCost Cost = 0;
+  VPCostContext CostCtx(CM.TTI, *CM.TLI, CM.Legal->getWidestInductionType(), CM,
+                        CM.CostKind);
+  LLVM_DEBUG(
+      dbgs() << "Calculating cost of work in vector early exit block:\n");
+  for (auto *ExitVPBB : Plan.getExitBlocks()) {
+    for (auto *PredVPBB : ExitVPBB->getPredecessors())
+      if (PredVPBB != Plan.getMiddleBlock())
```

Comment:
Could you add a comment explaining this step? I was not clear until I found: https://llvm.org/docs/Vectorizers.html#early-exit-vectorization, where it seems like this is looking for `vector.early.exit` from `early.exit` (maybe just link to this diagram).

Reply:
Done, hope the comments explain it better!

Reply:
Looks good, thanks :+1: